net: ip: Configurable IPv4 MTU

pdgendt opened the pull request:
Allow to configure the initial IPv4 MTU needed to support packets without fragmentation.

pdgendt committed:
net: ip: Configurable IPv4 MTU

Allow to configure the initial IPv4 MTU needed to support packets without
fragmentation.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

kartben commented on subsys/net/ip/Kconfig.ipv4:
```diff
@@ -42,6 +42,14 @@ config NET_IPV4_DEFAULT_NETMASK
 
 if NET_NATIVE_IPV4
 
+config NET_IPV4_MTU
```
please add to release notes so that we don't forget later :)

pdgendt replied:
Done

pdgendt committed:
doc: releases: releases-notes: 4.2: Add CONFIG_NET_IPV4_MTU entry

Add a release note entry for the newly introduced networking option
CONFIG_NET_IPV4_MTU.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt commented on include/zephyr/net/net_ip.h:
```diff
@@ -496,7 +496,11 @@ enum net_ip_mtu {
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	NET_IPV4_MTU = CONFIG_NET_IPV4_MTU,
+#else
 	NET_IPV4_MTU = 576,
```
Yes, the `config NET_NATIVE_IPV4` is a dependency for this option. This is done to be the same mechanism as IPV6.
